web/src/click_handlers.js

```diff
@@ -186,6 +187,15 @@ export function initialize() {
             return;
         }
 
+        if (narrow_state.filter()?.is_search()) {
+            // Clicks on a message from search results should bring the
+            // user to the message's near view instead of opening the
+            // compose box.
+            const msg = message_lists.current.selected_message();
+            window.location = hash_util.by_conversation_and_time_url(msg);
+            return;
+        }
+
```

Comment:
I wasn't sure if I should also try wrapping the element in a `<a>` tag or something, but decided against it.

Asked on CZO here https://chat.zulip.org/#narrow/stream/6-frontend/topic/clicking.20on.20search.20message/near/1583903